gradle/eclipse.gradle

```diff
@@ -1,23 +1,82 @@
-allprojects { project->
-	apply plugin: "eclipse"
-    plugins.withType(JavaPlugin) {
-        eclipse {
-            classpath {
-                file.whenMerged { classpath ->
-                    //There are classes in here not designed to be compiled, but just used in our testing
-                    classpath.entries.removeAll { it.path.contains('src/integTest/resources') }
-                    //Workaround for some projects referring to themselves as dependent projects
-                    classpath.entries.removeAll { it.path.contains("$project.name") && it.kind=='src' }
-                    classpath.entries.removeAll { it.path.contains("$project.name/build") && it.kind=='lib' }
-                }
-            }
-            jdt {
-                file.withProperties { properties ->
-                    //Eclipse's view of projects treat circular dependencies as errors by default
-                    //Treat these as warnings for now
-                    properties["org.eclipse.jdt.core.circularClasspath"] = "warning"
-                }
-            }
-        }
-    }
+// This is sort of a hack
+// 1. It will make most of gradle compile in eclipse
+// 2. It requires some adjustments in eclipse to work properly - see below
+//
+// Things required for this hack to work:
+//  1. Eclipse must be a recent one and needs the Groovyclipse feature. Take it from here, for example http://dist.springsource.org/snapshot/GRECLIPSE/e4.4/
+//  2. 2GB of memory is required (Add or alter -Xmx2g to eclipse.ini)
+//  3. The complete gradle project from github must be cloned (i.e. https://github.com/gradle/gradle)
+//  5. Run ./gradlew build on that folder
+//  6. Run ./gradlew eclipse on that folder
+//  7. Configure a JRE in Eclipse which is at least 1.7. Set the JRE name property of that JRE to 'JavaSE-1.5'
+//  8. In Window->Preferences->Groovy->Compiler, check 'Enable Script folder support' and Add '**/*.gradle'
+//  9. Import the gradle project into the Eclipse workspace, choose 'search for nested projects', take ALL projects
+// 10. Wait 5 minutes for Eclipse to calm down
+// 11. Accept that some things just don't seem to work in Eclipse, especially some of the groovy files
+
+import org.gradle.util.GUtil
+
+def subprojectsPath = new File(projectDir,'/subprojects').absolutePath
+
+apply plugin: 'eclipse'
+
+eclipse.project.file.withXml {
```

Comment:
This should not be necessary, as the root project does not have the Groovy plugin applied, so there should not be any builder. Even if there was, it should not have any source directories.

Can you make sure that you didn't have any old configuration left over? Maybe run `cleanEclipse eclipse`